### PR TITLE
librats: download hsk_cek cert to buffer other than local filesystem

### PR DESCRIPTION
**Pull request by wojiaohanliyang**
Avoid download hsk_cek cert to local filesystem

### CODE REVIEW

File: attesters/csv/collect_evidence.c [hunk 1]
```diff
@@ -12,6 +12,7 @@
 #include <librats/log.h>
 #include <librats/attester.h>
 #include <librats/csv.h>
+#include <curl/curl.h>
 #include "csv_utils.h"
 #include "../../verifiers/sev/sev_utils.c"
```
Comment (zeuson0) [inline attesters/csv/collect_evidence.c:17]: No need to include `sev_utils.c`

File: attesters/csv/collect_evidence.c [hunk 2]
```diff
@@ -81,6 +82,19 @@ static int do_hypercall(unsigned int p1, unsigned long p2, unsigned long p3)
...
+
+	if (realsize != HYGON_HSK_CEK_CERT_SIZE) {
+		RATS_ERR("hsk_cek size is not equal to %d bytes.", HYGON_HSK_CEK_CERT_SIZE);
+		return 0;
+	}
+	memcpy(userp, contents, realsize);
+
+	return realsize;
+}
+
```
Comment (zeuson0) [inline attesters/csv/collect_evidence.c:97]: `curl_writefunc_callback` may called times, please refer to the usage of this in `sev-snp` to modify it.

**Review by haosanzi — APPROVED** [review]

File: attesters/csv/collect_evidence.c [hunk 3]
```diff
@@ -81,6 +82,19 @@ static int do_hypercall(unsigned int p1, unsigned long p2, unsigned long p3)
 	return (int)ret;
 }
 
+static size_t curl_writefunc_callback(void *contents, size_t size, size_t nmemb, void *userp)
```
Comment (zeuson0) [inline attesters/csv/collect_evidence.c:85]: `curl_writefunc_callback` may called times, please refer to the usage of this in `sev-snp` to modify it.

Comment (wojiaohanliyang) [reply attesters/csv/collect_evidence.c:84]: Do you mean the macro definition of CURL_RETRY_TIMES in sev-snp/collect_evidence.c，I fix the retry times as 5 in csv attester

Comment (zeuson0) [reply attesters/csv/collect_evidence.c:84]: I mean you need to  refer to the implementation of `curl_writefunc_callback` in sev-snp/collect_evidence.c.

**Review by zeuson0 — APPROVED** [review]

**Pull request merged**